notification/store.go

```diff
@@ -253,54 +215,35 @@ func (db *DB) SendContactMethodVerification(ctx context.Context, id string, rese
 		return err
 	}
 	if rows != 1 {
-		return validation.NewFieldError("ContactMethod", "test message rate-limit exceeded")
+		return validation.NewFieldError("ContactMethod", fmt.Sprintf("Too many messages! Please try again in %.0f minute(s)", minTimeBetweenTests.Minutes()))
 	}
 
-	vID := uuid.NewV4().String()
+	vcID := uuid.NewV4().String()
 	code := db.rand.Intn(900000) + 100000
-	_, err = tx.Stmt(db.setVerificationCode).ExecContext(ctx, vID, id, code, fmt.Sprintf("%f seconds", (15*time.Minute).Seconds()), resend)
+	_, err = tx.StmtContext(ctx, db.setVerificationCode).ExecContext(ctx, vcID, cmID, code)
 	if err != nil {
 		return errors.Wrap(err, "set verification code")
 	}
 
 	return tx.Commit()
 }
 
-func (db *DB) VerifyContactMethod(ctx context.Context, cmID string, code int) ([]string, error) {
-	userID, err := db.cmUserID(ctx, cmID)
-	if err != nil {
-		return nil, err
-	}
-
-	tx, err := db.db.BeginTx(ctx, nil)
-	if err != nil {
-		return nil, err
-	}
-	defer tx.Rollback()
-
-	var cmValue string
-	err = db.verifyVerificationCode.QueryRowContext(ctx, cmID, code).Scan(&cmValue)
-	if err == sql.ErrNoRows {
-		return nil, validation.NewFieldError("Code", "unrecognized code")
+func (db *DB) VerifyContactMethod(ctx context.Context, cmID string, code int) error {
+	res, err := db.verifyAndEnableContactMethod.ExecContext(ctx, cmID, code)
```

Comment:
This still needs the call to `db.cmUserID()` -- you can throw away the user ID into `_` but it still performs validation and permissions checks we need.